daemons/compute/lib-cpp/CMakeLists.txt

```diff
@@ -42,45 +51,55 @@ add_custom_command(
         "${hw_proto}"
       DEPENDS "${hw_proto}")
 
-# Include generated *.pb.h files
-include_directories("${CMAKE_CURRENT_BINARY_DIR}")
+#
+# nnfdm library: libnnfdm.so for shared library and libnnfdm.a for static
+#
+if(BUILD_SHARED_LIBS)
+  add_library(nnfdm SHARED ${generated_srcs} client.cc client.h)
+else()
+  add_library(nnfdm STATIC ${generated_srcs} client.cc client.h)
+endif()
+
+#
+# Compile with position independant code if we are set to run as a shared library
+#
+set_target_properties(nnfdm PROPERTIES POSITION_INDEPENDANT_CODE ${BUILD_SHARED_LIBS})
```

Comment:
should this be POSITION_INDEPENDENT_CODE ?

Reply:
Nice catch!  This was left in by mistake (by me).  I'll take it it out.